===================================================================
RCS file: /cvs/gstreamer/gst-ffmpeg/ext/ffmpeg/gstffmpegdec.c,v
rcsdiff: /cvs/gstreamer/gst-ffmpeg/ext/ffmpeg/gstffmpegdec.c,v: warning: Unknown phrases like `commitid ...;' are present.
retrieving revision 1.94
retrieving revision 1.95
diff -u -r1.94 -r1.95
--- gstffmpegdec.c	2005/04/02 10:01:20	1.94
+++ gstffmpegdec.c	2005/04/14 10:50:15	1.95
@@ -778,14 +778,15 @@
 
   switch (GST_EVENT_TYPE (event)) {
     case GST_EVENT_EOS: {
-      gint have_data, len;
+      gint have_data, len, try = 0;
 
+      /* max. 10 times, for safety (see e.g. #300200) */
       do {
         len = gst_ffmpegdec_frame (ffmpegdec, NULL, 0, &have_data,
             &ffmpegdec->next_ts);
         if (len < 0 || have_data == 0)
           break;
-      } while (1);
+      } while (try++ < 10);
       goto forward;
     }
     case GST_EVENT_FLUSH:
